std/experimental/perpetual/perpetual.d

```diff
@@ -0,0 +1,192 @@
+module std.perpetual;
+
+private import std.mmfile;
+private import std.exception;
+private import std.conv : to;
+private import std.file : exists;
+private import std.traits : hasIndirections;
+
+
+
+/**
+ * Persistently maps value type object to file
+ */
+struct Perpetual(T)
+{
+	private MmFile _heap;
+	private enum _tag="Perpetual!("~T.stringof~")";
+
+	static if(is(T == Element[],Element))
+	{
+	// dynamic array 
+		private Element[] _value;
+		enum bool dynamic=true;
+		static assert(!hasIndirections!Element, Element.stringof~" is reference type");
+		@property Element[] Ref() { return _value; }
+		string toString() { return to!string(_value); }
+
+	}
+	else
+	{
+	// value type
```

Comment:
On Thu, 8/13/15, Dmitry Olshansky <notifications@github.com> wrote:
> the name most likely gonig to change as persistence is more general then mmap-ing objects 
>  to files.
Just accepted advise, I'm for persistent myself